### PR TITLE
add functional to add movies

### PR DESCRIPTION
**Pull request by mr-olesha**
https://olegpaunov.github.io/react_movies-list-add-form/

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+  const addMovies = (newMovie: Movie) => {
+    setMovies(PreviousMovies => ([...PreviousMovies, newMovie]));
```
Comment (roman-mirzoian) [inline src/App.tsx:11]:
```suggestion
    setMovies(prevMovies => ([...prevMovies, newMovie]));
```

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,96 @@
...
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
```
Comment (roman-mirzoian) [inline src/components/NewMovie/NewMovie.tsx:15]: All this should be kept in one state.

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,96 @@
...
+      imdbId,
+    };
+
+    onAdd(newMovie);
+
+    setCount(prevCount => prevCount + 1);
+    resetFormValues();
+  };
+
+  const isButtonEnabled = (
```
Comment (roman-mirzoian) [inline src/components/NewMovie/NewMovie.tsx:42]:
```suggestion
  const isAddButtonEnabled = (
```

**Review by roman-mirzoian — CHANGES_REQUESTED** [review]
Good job, but there is quite a bit of work to be done.

File: src/App.tsx (hunk 1 above)
Comment (BudnikOleksii) [reply src/App.tsx:11]: > 

Don't forget to check all comments

File: src/components/NewMovie/NewMovie.tsx [hunk 4]
```diff
@@ -1,45 +1,102 @@
...
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [state, setState] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
```
Comment (BudnikOleksii) [inline src/components/NewMovie/NewMovie.tsx:17]: You can create a default object for this state, pass it as an initial value, and then when you reset the form.

Also, the name `state` is not informative at all, what state?

File: src/components/NewMovie/NewMovie.tsx [hunk 5]
```diff
@@ -1,45 +1,102 @@
...
+
+    const newMovie: Movie = {
+      title: state.title,
+      description: state.description,
+      imgUrl: state.imgUrl,
+      imdbUrl: state.imdbUrl,
+      imdbId: state.imdbId,
+    };
+
+    onAdd(newMovie);
```
Comment (BudnikOleksii) [inline src/components/NewMovie/NewMovie.tsx:47]:
```suggestion
    onAdd(state);
```
You can just pass the state to `onAdd`, you don't need to create a `newMovie` object as it's the same

File: src/components/NewMovie/NewMovie.tsx [hunk 6]
```diff
@@ -1,45 +1,102 @@
...
+
+  const isAddButtonEnabled
+  = state.title.trim()
+    && state.imdbUrl.trim()
+    && state.imgUrl.trim()
+    && state.imdbId.trim();
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={handleSubmit}>
```
Comment (BudnikOleksii) [inline src/components/NewMovie/NewMovie.tsx:60]:
```suggestion
    <form
      className="NewMovie"
      key={count}
      onSubmit={handleSubmit}
    >
```
Don't forget about code style

**Review by BudnikOleksii — CHANGES_REQUESTED** [review]
Good job 👍 
Just a few things need to be fixed

**Review by BudnikOleksii — APPROVED** [review]
Good work 👍

**Pull request closed without merging**